airflow/models.py

```diff
@@ -2463,6 +2463,29 @@ def get_many(
         return query.all()
 
 
+class DagRun(Base):
+    """
+    DagRun describes an instance of a Dag. It can be created
+    by the scheduler (for regular runs) or by an external trigger
+    """
+    __tablename__ = "dag_run"
+
+    dag_id = Column(String(ID_LEN), primary_key=True)
+    execution_date = Column(DateTime, primary_key=True)
+    run_id = Column(String(ID_LEN))
+    external_trigger = Column(Boolean, default=False)
```

Comment:
I would use a string to be able to mark failed dags as well.

I think it is quite helpful to have states on dag level (active, success, failed). But up to now there is no mechanism to know when a dag really has finished as all tracking is done on task instance level. After each task instance run the scheduler would have to check whether this was the last task to be run. Will think about that.